File: 3. Recursion/C++/8. Nth fibonacci number.cpp
/*
Nth Fibonaaci number using Recursion
*/

#include <bits/stdc++.h>
using namespace std;

//Time complexity = 2^n
int fib(int n){
    if(n==1 || n==0){
        return n;
    }
    return fib(n-1)+fib(n-2);
}

int main() {
	// your code goes here
    int n;
    cin >> n;
    int ans = fib(n);
    cout << "Nth Fibonaaci number is " << ans;
	return 0;
}

File: 3. Recursion/Python/8. Nth fibonacci number.py
# cook your dish here
#Nth Fibonaaci number using Recursion

#Time complexity = 2^n
def fib(n):
    if(n==1 or n==0):
        return n
    return fib(n-1)+fib(n-2)
    
n = int(input())
ans = fib(n)
print(ans)
